sds_data_manager/lambda_code/SDSCode/pointing_frame_handler.py

```diff
@@ -0,0 +1,248 @@
+"""Generate Pointing Frame.
+
+Notes
+-----
+Kernels that are required to run this code:
+1. imap_science_0001.tf - pointing frame kernel
+2. imap_sclk_0000.tsc - spacecraft clock kernel
+3. imap_wkcp.tf - spacecraft frame kernel
+4. de430.bsp - standard SPICE planetary ephemeris kernel
+5. naif0012.tls - standard NAIF leapsecond kernel
+6. imap_spin.bc - test attitude kernel available at:
+https://lasp.colorado.edu/galaxy/display/IMAP/Data
+This needs to be placed in tests/test-data/spice/imap_spin.bc
+
+References
+----------
+https://spiceypy.readthedocs.io/en/main/documentation.html.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+import numpy as np
+import spiceypy as spice
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# Each spin is 15 seconds. We want 10 quaternions per spin.
+# duration / # samples (nominally 15/10 = 1.5 seconds)
+STEP = 1.5
+
+
+def get_coverage(ck_kernel):
+    """Create the pointing frame.
+
+    Parameters
+    ----------
+    ck_kernel : str
+        Path of ck_kernel used to create the pointing frame.
+
+    Returns
+    -------
+    et_start : float
+        Start time of ck_kernel.
+    et_end : float
+        End time of ck_kernel.
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+    """
+    # Get the spacecraft ID.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.gipool
+    id_imap_spacecraft = spice.gipool("FRAME_IMAP_SPACECRAFT", 0, 1)
+
+    # TODO: Queried pointing start and stop times here.
+
+    # Get the coverage window
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.ckcov
+    cover = spice.ckcov(ck_kernel, int(id_imap_spacecraft), True, "SEGMENT", 0, "TDB")
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.wnfetd
+    et_start, et_end = spice.wnfetd(cover, 0)
+    et_times = np.arange(et_start, et_end, STEP)
```

Comment:
`arange` can produce sometimes odd results when used with a float input. I'd suggest using `linspace` instead.

From the docs: https://numpy.org/doc/stable/reference/generated/numpy.arange.html
> When using a non-integer step, such as 0.1, it is often better to use [numpy.linspace](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html#numpy.linspace).

```suggestion
    et_times = np.linspace(et_start, et_end, (et_end - et_start) // 10)
```

Reply:
Odd. I got an error trying to use a non-integer step and linspace.

Reply:
Yes, it isn't a non-integer step then, it is the integer number of items you want. The code I suggested provides an integer with the `//` operator.